### ISSUE TITLE
cosineDistance w/ Float

### ISSUE THREAD

**Issue by eggie5** [scalanlp/breeze#818]
`cosineDistance` doesn't seem to work on `DenseVector[Float]`. If I change to `Double` it works.

```
  val v1 = DenseVector(1.0f, 2.0f)
  val v2 = DenseVector(3.0f, 4.0f)
  println(cosineDistance(v1, v2))
```

```
diverging implicit expansion for type breeze.generic.UFunc.UImpl2[breeze.linalg.functions.cosineDistance.type,breeze.linalg.Vector[Float],breeze.linalg.Vector[Float],VR]
starting with method collapseUred in object UFunc
  println(cosineDistance(v1, v2))
```

Is this expected?

**Comment by dlwh** [comment]
this is the kind of thing that should work. I'll try to fix it for the next
2.0 RC

**Comment by dlwh** [comment]
fixed in 5828a7c584887bada9dc4a809b085059ec4b06b8
